tools/doc/generate.js

```diff
@@ -61,33 +61,37 @@ args.forEach((arg) => {
 nodeVersion = nodeVersion || process.version;
 
 if (!filename) {
-  throw new Error('No input file specified');
+  throw new Error("No input file specified");
 } else if (!outputDir) {
-  throw new Error('No output directory specified');
+  throw new Error("No output directory specified");
 }
 
+const fileTask = async () => {
+  try {
+    const input = await fs.readFile(filename, "utf8");
+    const content = unified()
+      .use(markdown)
+      .use(html.preprocessText)
+      .use(json.jsonAPI, { filename })
+      .use(html.firstHeader)
+      .use(html.preprocessElements, { filename })
+      .use(html.buildToc, { filename, apilinks })
+      .use(remark2rehype, { allowDangerousHTML: true })
+      .use(raw)
+      .use(htmlStringify)
+      .process(input);
 
-fs.readFile(filename, 'utf8', async (er, input) => {
-  if (er) throw er;
-
-  const content = unified()
-    .use(markdown)
-    .use(html.preprocessText)
-    .use(json.jsonAPI, { filename })
-    .use(html.firstHeader)
-    .use(html.preprocessElements, { filename })
-    .use(html.buildToc, { filename, apilinks })
-    .use(remark2rehype, { allowDangerousHTML: true })
-    .use(raw)
-    .use(htmlStringify)
-    .processSync(input);
-
-  const basename = path.basename(filename, '.md');
-
-  const myHtml = await html.toHTML({ input, content, filename, nodeVersion });
-  const htmlTarget = path.join(outputDir, `${basename}.html`);
-  fs.writeFileSync(htmlTarget, myHtml);
-
-  const jsonTarget = path.join(outputDir, `${basename}.json`);
-  fs.writeFileSync(jsonTarget, JSON.stringify(content.json, null, 2));
-});
+    const basename = path.basename(filename, ".md");
+    const myHtml = await html.toHTML({ input, content, filename, nodeVersion });
+    const htmlTarget = path.join(outputDir, `${basename}.html`);
+    const data = await fs.writeFile(htmlTarget, myHtml);
```

Comment:
you don't need to assign to a variable with await, no need for these data variables.